src/err.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2016 Jonathan 'theJPster' Pallant.
+//
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+//
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHORS DISCLAIM ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY
+// SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+// OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+// CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+//! Implements types influenced by `std::io::Result` for indicating success or failure
+//! of a routine.
+
+use std::result;
+
+/// Used by routines that return a certain type if they pass, or
+/// wish to indicate failure without any sense of why they failed.
+pub type Result<T> = result::Result<T, ()>;
+
+/// Used by routines that wish to indicate success or failure, with
+/// no associated value either way.
+pub type EmptyResult = Result<()>;
```

Comment:
OK, I think we're about as close to agreement as we're going to get on the basic issue that we shouldn't use `()` for the error type. I pushed to master a change that replaces `()` with `ring::error::Unspecified`. This is not exactly what you suggested but actually it is not too far off.

If you still want to try to clarify the implementation of the code by having some kind of aliases for `Result<(), error::Unspecified>` and/or `Err(error::Unspecified)` and/or `Ok(())`, I am open to discussing that. But, that should probably happen either in the issue or in another PR.

Thanks for pushing for these improvements!